### PR TITLE
Allow solver to provide claimed objective value (gas savings)

### LINKED ISSUE #221: [PoC] Work with claimed proposed objective value
In order to prevent the very long calculations for the objective value, we could check for the best solution much quicker by working with a claimed objective value.

Then we can at the beginning we can check for a better objective value with the claimed value and later check that the claimed value is the actual value

### PR DESCRIPTION
**Pull request by bh2smith**
A first pass through a new parameter on `submitSolution` which is meant as a gas saving tool. That is, the solver can submit their claimed objective value which is first compared with the contract's existing solution before any further computation is performed. I have decided against an additional requirement at the end (which would ensure the claimedObjective equals the evaluated objective) because it seems slightly unnecessary to introduce a different logical branch (as this is meant as a gas saving service for solvers).

This is still in draft format because it remains to compute these hardcoded numbers in the unit tests. Furthermore, this will also require adjustments after we have fixed the disregarded utility bug #276 

**Please use the following squash-commit message**

To prevent lengthy objective value calculations, contract compares claimed objective value with current. 
Main purpose avoid race condition where a solver thought they had the highest scoring solution but in between (e.g. earlier in the same block), someone else submitted a better result.
Note that there is no additional requirement verifying that claimedObjective equals the evaluated objective since it seems slightly unnecessary to introduce a different logical branch (as this is meant as a gas saving service for solvers).

Closes #221

### CODE REVIEW

File: test/stablecoin_converter.js [hunk 1]
```diff
@@ -199,8 +199,9 @@ contract("StablecoinConverter", async (accounts) => {
       const orderId = [orderId1, orderId2]
       const tokenIdsForPrice = basicTrade.solution.tokenIdsForPrice
 
+      // Note that we claim a better solution than we have...
       await truffleAssert.reverts(
-        stablecoinConverter.submitSolution(batchIndex, owner, orderId, [0, 0], prices, tokenIdsForPrice, { from: solutionSubmitter }),
+        stablecoinConverter.submitSolution(batchIndex, 1, owner, orderId, [0, 0], prices, tokenIdsForPrice, { from: solutionSubmitter }),
```
Comment (josojo) [inline test/stablecoin_converter.js]: I would prefer to calculate the claimed values.

Then refactoring later the test is much better.

Comment (bh2smith) [reply test/stablecoin_converter.js]: Yes. You will notice that this is a todo in the Draft PR description.

**Review by josojo — APPROVED** [review]
This was quick! great!

**Review by nlordell — APPROVED** [review]

**Comment by josojo** [comment]
Please don't merge before we have adapted the rust backend

**Pull request merged**